pkg/cloud/aws/actuators/scope.go

```diff
@@ -34,6 +35,29 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/patch"
 )
 
+var (
+	sessionCache sync.Map
+)
+
+func sessionForConfig(config *aws.Config) (*session.Session, error) {
```

Comment:
Should this be `sessionForRegion` and only take a `Region` parameter? The current naming and param seems misleading, since if I called it with a different Config, but it had the same Region as an already cached Session, I would get a Session that doesn't necessarily match my config.

Reply:
I went back and forth on it, I think down the line we might want to change the hashing function from other config params

Reply:
Let's start with what we're actively using now - sessionForRegion. We can change later, but only if needed.